deepchecks/nlp/checks/train_test_validation/train_test_sample_mix.py

```diff
@@ -0,0 +1,150 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains train-test samples mix check."""
+import typing as t
+import pandas as pd
+
+from deepchecks.core import CheckResult
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp import Context, TrainTestCheck
+from deepchecks.utils.strings import format_percent
+from deepchecks.nlp.utils.text_utils import normalize_text, hash_text
+from deepchecks.utils.strings import format_list, format_percent
+from deepchecks.utils.abstracts.train_test_samples_mix import TrainTestSamplesMixAbstract
+from deepchecks.nlp.text_data import TextData
+from deepchecks.nlp.checks.data_integrity.text_duplicates import to_ordional_enumeration
```

Comment:
if after refactor still used in both checks move it to utils

Reply:
moved